pkg/k8s/apis/cilium.io/v2/types.go

```diff
@@ -260,7 +260,9 @@ func (r *CiliumNetworkPolicy) Parse() (api.Rules, error) {
 	if r.Spec != nil {
 		if err := r.Spec.Sanitize(); err != nil {
 			return nil, fmt.Errorf("Invalid CiliumNetworkPolicy spec: %s", err)
-
+		}
+		if namespace != "" && r.Spec.NodeSelector.LabelSelector != nil {
+			return nil, fmt.Errorf("NodeSelector is only valid in CiliumClusterwideNetworkPolicy")
```

Comment:
For reviewers: I'm not super sure this is the right place to have this check. Maybe it should be checked in `register.go` instead?

Reply:
Ideally, this is something that we'd want to do in the validation schemas. But since CCNP is basically CNP, but without the namespace scope, I don't think it's possible to have logic like this in the validation schema. 

This seems OK, but I'd like @aanm's opinion as well.

Reply:
Should I move this to the validation schemas then? As far as I understand, I'd just need to have separate schemas for CCNP and CNP, right?

Reply:
Correct.

Reply:
Done. @aanm Please take a look.